app/js/views/service-tags.jsx

```diff
@@ -0,0 +1,98 @@
+import React from 'components/react';
+
+import BaseView from 'js/views/base-view';
+import TagList from 'js/views/tag-list';
+
+export default class ServiceTags extends BaseView {
+  constructor(props) {
+    super(props);
+
+    this.state = { service: null, tags: [] };
+    this.foxbox = props.foxbox;
+
+    this.onServiceStateChanged = this.onServiceStateChanged.bind(this);
+  }
+
+  componentDidMount() {
+    this.foxbox.services.get(this.props.id)
+      .then((service) => {
+        this.setState({ service, tags: service.tags });
+      })
+      .catch((error) => {
+        console.error('Error occurred while retrieving service: ', error);
+      });
+
+    this.foxbox.services.on('service-changed', this.onServiceStateChanged);
+  }
+
+  componentWillUnmount() {
+    this.foxbox.services.off('service-changed', this.onServiceStateChanged);
+  }
+
+  onServiceStateChanged(service) {
+    if (service.id !== this.props.id) {
+      return;
+    }
+
+    this.setState({ service, tags: service.tags });
+  }
+
+  onAddTag() {
+    const service = this.state.service;
+    if (!service) {
+      return;
+    }
+
+    const tag = prompt('Enter new tag name');
+    if (!tag || !tag.trim()) {
+      return;
+    }
+
+    service.addTag(tag.trim())
+      .catch((err) => {
+        console.error('Could not add a new tag: %o', err);
+      })
+      .then(() => {
+        this.setState({ tags: service.tags });
```

Comment:
This is better to do optimist updates.
You set the state immediately before doing the request to the box and if anything go wrong, then you undo the change.
This gives suprisingly a fast feeling to the UI.